### PR TITLE
DAC overlays 

### PR DESCRIPTION
**Pull request by pifi-bz**
- as discussed.

### CODE REVIEW

File: arch/arm/boot/dts/overlays/Makefile [hunk 1]
```diff
@@ -130,6 +130,8 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	pibell.dtbo \
 	pifacedigital.dtbo \
 	pifi-40.dtbo \
+	pifi-dac-zero.dtbo \
+	pifi-dac-hd.dtbo \
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/Makefile:133]: Alphabetical order, please.

File: arch/arm/boot/dts/overlays/README [hunk 2]
```diff
@@ -2131,6 +2131,18 @@ Load:   dtoverlay=pifi-40
 Params: <None>
 
 
+Name:   pifi-dac-zero
+Info:   Configures the PiFi DAC Zero
+Load:   dtoverlay=pifi-dac-zero
+Params: <None>
+
+
+Name:   pifi-dac-hd
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/README:2134]: Order again.

File: arch/arm/boot/dts/overlays/pifi-dac-hd-overlay.dts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+// Overlay for PiFi-DAC-HD
+/dts-v1/;
+/plugin/;
+
+/ {
+	compatible = "brcm,bcm2708";
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/pifi-dac-hd-overlay.dts]: Please use `"brcm,bcm2835"`.

File: arch/arm/boot/dts/overlays/pifi-dac-hd-overlay.dts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	};
+
+	fragment@2 {
+		target = <&sound>;
+		__overlay__ {
+			compatible = "simple-audio-card";
+			simple-audio-card,name = "PiFi-DAC-HD";
+			status="okay";
+
+			playback_link: simple-audio-card,dai-link@1 {
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/pifi-dac-hd-overlay.dts]: FYI, if nothing references a label (e.g. `&playback_link`) in an overlay and it isn't exported then it serves no purpose. The same goes for `p_cpu_dai` and `p_codec_dai`.

File: arch/arm/boot/dts/overlays/pifi-dac-hd-overlay.dts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+			};
+		};
+	};
+
+	fragment@2 {
+		target = <&sound>;
+		__overlay__ {
+			compatible = "simple-audio-card";
+			simple-audio-card,name = "PiFi-DAC-HD";
+			status="okay";
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/pifi-dac-hd-overlay.dts]: It's a bit picky, but the rest of the overlay is so nicely formatted that the lack of spaces here stands out.

File: arch/arm/boot/dts/overlays/pifi-dac-zero-overlay.dts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    fragment@0 {
+        target = <&sound>;
+        __overlay__ {
+            compatible = "simple-audio-card";
+            simple-audio-card,name = "PiFi-DAC-Zero";
+
+            status = "okay";
+
+            playback_link: simple-audio-card,dai-link@1 {
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/pifi-dac-zero-overlay.dts]: Same comment about unused labels.

File: arch/arm/boot/dts/overlays/pifi-dac-hd-overlay.dts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+/plugin/;
+
+/ {
+	compatible = "brcm,bcm2708";
+
+	fragment@0 {
+		target = <&i2s>;
+		__overlay__ {
+			status = "okay";
+        };
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/pifi-dac-hd-overlay.dts]: This line looks like a mixture of spaces and TABs - not a crime (unlike in the README), just a bit ugly.

**Review by pelwell — COMMENTED** [review]
I got my wish with finding something to comment on, but it's all minor stuff.

**Comment by pifi-bz** [comment]
Ooh dear.  These should be all tidied up in the latest commit - are you OK with said commit being added to this pull request, or should I redo from start (as my Commodore PET used to say..)

**Comment by pelwell** [comment]
If you are creating these commit in the GitHub GUI then do it any way you like and I'll squash them all. If you're creating the PRs on the command line and pushing them then the general way to update a PR is:
1. Add a new commit with the changes. If the PR has multiple commits and you want to change more than one then split your changes across as many new commits. `git add -p` is useful for selectively staging changes in preparation for creating commits out of some of them.
2.  Use `git rebase -i HEAD~<n>`, where `<n>` is the number of commits at the top of the branch that should be considered for the PR. Having an `<n>` that is too large normally just slows things down, but it gets ugly if one of the extras is a merge commit...
3. Use `git push -f origin HEAD:<remote-branch-name>` to update the PR.

**Comment by pifi-bz** [comment]
How's that?

File: arch/arm/boot/dts/overlays/pifi-dac-hd-overlay.dts [hunk 8]
```diff
@@ -3,23 +3,23 @@
 /plugin/;
 
 / {
-	compatible = "brcm,bcm2708";
+	compatible = "brcm,bcm2735";
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/pifi-dac-hd-overlay.dts]: Oooh - so close.

Comment (pifi-bz) [reply arch/arm/boot/dts/overlays/pifi-dac-hd-overlay.dts]: Ouch.  Thank you, again, for your patience.

**Pull request merged**